test/Prop.hs

```diff
@@ -990,6 +1000,55 @@ monadBind constr eq t (a, b) = withMaxSuccess maxTestCount $
         let list = a >>= \x -> (+ x) <$> b
         listEquals eq stream list
 
+-------------------------------------------------------------------------------
+-- tests for S.groupsBy
+-------------------------------------------------------------------------------
+
+-- returns the integer with minimum value from the [Int]
+minl :: Int -> [Int] -> Int
+minl curr [] = curr
+minl curr (x:xl) = minl (min curr x) xl
+
+
+-- if list is empty, returns Nothing
+-- else wraps minimum value of [Int] in Just
+mayBeMinl :: [Int] -> Maybe Int
+mayBeMinl [] = Nothing
+mayBeMinl (x:xl) = Just $ minl x xl
+
+
+-- after grouping (and folding) Int stream using (>) operation,
+-- the first Int of every [Int] (in the [Int] stream) should be the minimum
+testGroupsBy :: Property
+testGroupsBy =
+    forAll (choose (0, maxStreamLen)) $ \len ->
+            forAll (vectorOf len (arbitrary :: Gen Int)) $ \vec -> monadicIO $ do
+                a <- run $ S.all (\l -> case l of
+                                          [] -> True
+                                          (ex:_) -> ex == minl intMax l) $ S.groupsBy (>) FL.toList $ S.fromList vec
```

Comment:
You can just use `ex == minimum l` here. Also, the general convention to decompose a list is `x:xs` so you can use `x` instead of `ex`. Also the result value can be named as `r`, `res` or `result`.

Reply:
By result, do you mean `a` which should be `True`?

Reply:
yes